src/frontend/js/components/CourseAddToWishlist/index.tsx

```diff
@@ -0,0 +1,88 @@
+import { useMemo } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { useUserWishlistCourses } from 'hooks/useUserWishlistCourses';
+import type * as Joanie from 'types/Joanie';
+import { Button } from 'components/Button';
+import { Spinner } from 'components/Spinner';
+import { useSession } from 'data/SessionProvider';
+
+const messages = defineMessages({
+  labelAdd: {
+    id: 'components.CourseAddToWishlist.labelAdd',
+    description: 'Label to proceed to add a course to user wishlist.',
+    defaultMessage: 'Notify me',
+  },
+  labelRemove: {
+    id: 'components.CourseAddToWishlist.labelRemove',
+    description: 'Label to proceed to remove a course to user wishlist.',
+    defaultMessage: 'Do not notify me anymore',
+  },
+  logMe: {
+    id: 'components.CourseAddToWishlist.logMe',
+    defaultMessage: 'Log in to be notified',
+    description: 'Label to proceed to login page before being adding a course to user wishlist.',
+  },
+});
+
+enum ComponentStates {
+  INITIALIZING = 'initializing',
+  LOADING = 'loading',
+  IDLE = 'idle',
+  ERROR = 'error',
+}
+
+export interface Props {
+  courseCode: Joanie.Course['code'];
+}
+
+const CourseAddToWishlist = ({ courseCode }: Props) => {
+  const { user, login } = useSession();
+  const wishlistCourse = useUserWishlistCourses({ courseCode });
+
+  const componentState = useMemo<ComponentStates>(() => {
+    if (wishlistCourse.states.fetching) return ComponentStates.LOADING;
+    if (wishlistCourse.states.error) return ComponentStates.ERROR;
+    return ComponentStates.IDLE;
+  }, [wishlistCourse.states.fetching, wishlistCourse.states.error]);
+  const isInWishListCourse = useMemo(
+    () => wishlistCourse.items?.length > 0,
+    [wishlistCourse.items],
+  );
+
+  const removeFromWishlist = () => wishlistCourse.methods.delete(courseCode);
+  const addToWishlist = () => wishlistCourse.methods.create(courseCode);
+
+  if (user && !wishlistCourse.states.isFetched) return <Spinner />;
```

Comment:
It's not the same.
Here i want the spinner instead of the component in case initial data aren't fetched.

`isLoading` will switch between `true` and `false` each time request append. It's perfect for the disable props of `Button`, but not for the initial display of the component.